packages/common/src/hooks/usePagination/usePagination.ts

```diff
@@ -1,12 +1,15 @@
 import { useState, useEffect } from 'react';
 
-export interface PaginationState {
+interface Pagination {
   page: number;
   offset: number;
   first: number;
   onChangePage: (newPage: number) => void;
   onChangeFirst: (newFirst: number) => void;
 }
+export interface PaginationState extends Pagination {
```

Comment:
I'm not sure why the addition of the `pagination` object, as well as using the individual member properties.
There's a good reason there somewhere, I'm just not seeing it. can you help me out?

Reply:
Umm yeah it's a bit weird. It's so you can destructure one or many properties and also keep a 'group'.. somewhat nice  for composing with other hooks..